### PR TITLE
Migrate to ESLint 9

### PR DESCRIPTION
**Pull request by dabico**
Combining multiple dependencies PRs into one.

<details>
<summary>Instructions for merging</summary>

* **Use a merge commit**, so that GitHub will mark all original PRs as merged.
* If your repository does not have merge commits enabled, please temporarily enable them in settings. Tick `Allow merge commits` in the repository settings.
* When ready, merge this PR using `Create a merge commit`.

</details>

## Combined PRs

* Bump @typescript-eslint/eslint-plugin from 7.18.0 to 8.0.1 (#91) @app/dependabot
* Bump typescript-eslint from 7.18.0 to 8.0.1 (#92) @app/dependabot

### CODE REVIEW

File: .eslintignore
Comment (DaveKeehl) [inline .eslintignore:1]: Are we no longer ignoring any file when linting?

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "type": "module",
   "scripts": {
     "lint": "eslint .",
-    "format": "prettier . --write",
+    "format": "eslint --fix .",
```
Comment (DaveKeehl) [inline package.json:13]: Just out of curiosity, why the switch?

File: .eslintignore
Comment (dabico) [reply .eslintignore:1]: The `.eslintignore` has been deprecated as of version 9. Ignore patterns are now specified as part of the config. See: https://github.com/lettercrap/lettercrap/pull/93/files#diff-a32a0887ed9d1d707bbb3b845b7df7fd40e673c47e7b60a3ebd896b68d3b8839R13.

File: package.json (hunk 1 above)
Comment (dabico) [reply package.json:13]: The whole idea behind integrating Prettier into ESLint is for it to inherit its ruleset while allowing us to configure how these rules are applied. It's generally much better to call a single linter as opposed to maintaining potentially diverging configurations across multiple linters. For instance, if your `.prettierignore` was different from the files ignored by ESLint, then calling `prettier --write` and `eslint --fix` may produce different results.

**Review by DaveKeehl — APPROVED** [review]

**Pull request merged**